### ISSUE TITLE
why torch.nn.SiLU gets the different result, compared with input * torch.sigmoid(input)

### ISSUE THREAD

**Issue by yyynbone** [pytorch/pytorch#86781]
### 🐛 Describe the bug

I try to compare the silu and swish function, and find the result is different
```python
    import torch
    x = torch.randn((2,5,6))
    x2 = x.clone()
    x1_out = torch.nn.SiLU()(x)
    x2_out = x2 * torch.sigmoid(x2)
    print(x)
    print(x2)
    print(x1_out)
    print(x2_out)
    print('now we compare x and x2:\n')
    print(x == x2)
    print('now we compare the act result of x and x2:\n')
    print(x1_out == x2_out)
```

```
tensor([[[ 9.3163e-01, -5.3143e-01,  6.7229e-01, -8.7400e-01, -7.9338e-01,
           9.1866e-01],
         [ 1.3104e+00, -5.7266e-01,  3.6137e-01,  1.4891e+00,  1.2401e+00,
           3.6765e-01],
         [-1.1675e+00,  1.1988e-01, -8.8352e-01,  5.4856e-01,  6.8558e-01,
           2.7748e-01],
         [-7.7592e-02, -9.7061e-04,  9.2009e-01, -6.4594e-01,  1.6360e+00,
           1.3096e-01],
         [ 6.5638e-01,  1.2074e+00, -7.4757e-01, -6.0530e-02, -1.0654e+00,
           1.2091e+00]],

        [[ 2.5669e-01, -1.7922e+00,  1.1939e+00, -1.4033e+00,  8.9764e-01,
          -7.2533e-01],
         [ 6.1281e-01, -7.1519e-01, -5.3847e-01,  5.0693e-01,  5.4190e-02,
          -4.1832e-01],
         [ 1.5643e-01,  3.1940e-01, -8.3219e-01, -2.6669e-01, -2.5031e-01,
           4.5189e-03],
         [ 1.3848e+00, -1.4136e+00,  3.6532e-01, -1.1374e+00, -3.1367e-01,
          -1.0363e+00],
         [ 1.2315e+00, -6.9107e-01,  1.2036e+00,  5.5551e-02,  1.7043e+00,
          -4.5110e-01]]])
tensor([[[ 9.3163e-01, -5.3143e-01,  6.7229e-01, -8.7400e-01, -7.9338e-01,
           9.1866e-01],
         [ 1.3104e+00, -5.7266e-01,  3.6137e-01,  1.4891e+00,  1.2401e+00,
           3.6765e-01],
         [-1.1675e+00,  1.1988e-01, -8.8352e-01,  5.4856e-01,  6.8558e-01,
           2.7748e-01],
         [-7.7592e-02, -9.7061e-04,  9.2009e-01, -6.4594e-01,  1.6360e+00,
           1.3096e-01],
         [ 6.5638e-01,  1.2074e+00, -7.4757e-01, -6.0530e-02, -1.0654e+00,
           1.2091e+00]],

        [[ 2.5669e-01, -1.7922e+00,  1.1939e+00, -1.4033e+00,  8.9764e-01,
          -7.2533e-01],
         [ 6.1281e-01, -7.1519e-01, -5.3847e-01,  5.0693e-01,  5.4190e-02,
          -4.1832e-01],
         [ 1.5643e-01,  3.1940e-01, -8.3219e-01, -2.6669e-01, -2.5031e-01,
           4.5189e-03],
         [ 1.3848e+00, -1.4136e+00,  3.6532e-01, -1.1374e+00, -3.1367e-01,
          -1.0363e+00],
         [ 1.2315e+00, -6.9107e-01,  1.2036e+00,  5.5551e-02,  1.7043e+00,
          -4.5110e-01]]])
tensor([[[ 6.6836e-01, -1.9673e-01,  4.4507e-01, -2.5732e-01, -2.4709e-01,
           6.5663e-01],
         [ 1.0321e+00, -2.0651e-01,  2.1298e-01,  1.2150e+00,  9.6184e-01,
           2.1724e-01],
         [-2.7706e-01,  6.3527e-02, -2.5838e-01,  3.4768e-01,  4.5590e-01,
           1.5787e-01],
         [-3.7292e-02, -4.8507e-04,  6.5792e-01, -2.2214e-01,  1.3693e+00,
           6.9763e-02],
         [ 4.3219e-01,  9.2946e-01, -2.4023e-01, -2.9349e-02, -2.7304e-01,
           9.3119e-01]],

        [[ 1.4473e-01, -2.5594e-01,  9.1628e-01, -2.7686e-01,  6.3774e-01,
          -2.3662e-01],
         [ 3.9746e-01, -2.3491e-01, -1.9845e-01,  3.1637e-01,  2.7829e-02,
          -1.6604e-01],
         [ 8.4323e-02,  1.8499e-01, -2.5230e-01, -1.1567e-01, -1.0957e-01,
           2.2646e-03],
         [ 1.1075e+00, -2.7660e-01,  2.1566e-01, -2.7616e-01, -1.3244e-01,
          -2.7137e-01],
         [ 9.5326e-01, -2.3068e-01,  9.2579e-01,  2.8547e-02,  1.4420e+00,
          -1.7552e-01]]])
tensor([[[ 6.6836e-01, -1.9673e-01,  4.4507e-01, -2.5732e-01, -2.4709e-01,
           6.5663e-01],
         [ 1.0321e+00, -2.0651e-01,  2.1298e-01,  1.2150e+00,  9.6184e-01,
           2.1724e-01],
         [-2.7706e-01,  6.3527e-02, -2.5838e-01,  3.4768e-01,  4.5590e-01,
           1.5787e-01],
         [-3.7292e-02, -4.8507e-04,  6.5792e-01, -2.2214e-01,  1.3693e+00,
           6.9763e-02],
         [ 4.3219e-01,  9.2946e-01, -2.4023e-01, -2.9349e-02, -2.7304e-01,
           9.3119e-01]],

        [[ 1.4473e-01, -2.5594e-01,  9.1628e-01, -2.7686e-01,  6.3774e-01,
          -2.3662e-01],
         [ 3.9746e-01, -2.3491e-01, -1.9845e-01,  3.1637e-01,  2.7829e-02,
          -1.6604e-01],
         [ 8.4323e-02,  1.8499e-01, -2.5230e-01, -1.1567e-01, -1.0957e-01,
           2.2646e-03],
         [ 1.1075e+00, -2.7660e-01,  2.1566e-01, -2.7616e-01, -1.3244e-01,
          -2.7137e-01],
         [ 9.5326e-01, -2.3068e-01,  9.2579e-01,  2.8547e-02,  1.4420e+00,
          -1.7552e-01]]])
now we compare x and x2:

tensor([[[True, True, True, True, True, True],
         [True, True, True, True, True, True],
         [True, True, True, True, True, True],
         [True, True, True, True, True, True],
         [True, True, True, True, True, True]],

        [[True, True, True, True, True, True],
         [True, True, True, True, True, True],
         [True, True, True, True, True, True],
         [True, True, True, True, True, True],
         [True, True, True, True, True, True]]])
now we compare the act result of x and x2:

tensor([[[ True,  True,  True,  True,  True,  True],
         [ True,  True,  True, False,  True, False],
         [ True,  True,  True,  True,  True,  True],
         [ True,  True,  True,  True,  True,  True],
         [False,  True,  True,  True,  True,  True]],

        [[ True,  True, False,  True,  True,  True],
         [ True, False, False,  True,  True, False],
         [ True, False,  True,  True,  True, False],
         [ True,  True,  True,  True,  True,  True],
         [ True,  True, False, False,  True,  True]]])
```

### Versions

Collecting environment information...
PyTorch version: 1.11.0
Is debug build: False
CUDA used to build PyTorch: 11.3
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 11 Pro
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect
Libc version: N/A

Python version: 3.9.7 (default, Sep 16 2021, 16:59:28) [MSC v.1916 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.22000-SP0
Is CUDA available: True
CUDA runtime version: 11.5.50
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3080 Ti
Nvidia driver version: 496.49
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.20.3
[pip3] numpydoc==1.1.0
[pip3] torch==1.11.0
[pip3] torchaudio==0.11.0
[pip3] torchvision==0.12.0
[conda] blas                      1.0                         mkl    defaults
[conda] cudatoolkit               11.3.1               h59b6b97_2    https://mirrors.bfsu.edu.cn/anaconda/pkgs/main
[conda] mkl                       2021.4.0           haa95532_640    defaults
[conda] mkl-service               2.4.0            py39h2bbff1b_0    defaults
[conda] mkl_fft                   1.3.1            py39h277e83a_0    defaults
[conda] mkl_random                1.2.2            py39hf11a4ad_0    defaults
[conda] mypy_extensions           0.4.3            py39haa95532_0    defaults
[conda] numpy                     1.20.3           py39ha4e8547_0    defaults
[conda] numpy-base                1.20.3           py39hc2deb75_0    defaults
[conda] numpydoc                  1.1.0              pyhd3eb1b0_1    defaults
[conda] pytorch                   1.11.0          py3.9_cuda11.3_cudnn8_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/pytorch
[conda] pytorch-mutex             1.0                        cuda    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/pytorch
[conda] torchaudio                0.11.0               py39_cu113    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/pytorch
[conda] torchvision               0.12.0               py39_cu113    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/pytorch

**Comment by yyynbone** [comment]
> 

Really Appreciate!
